### DISCUSSION TITLE
ZK Sync Foundry & M1 Mac Issues

### DISCUSSION THREAD

**Discussion by JesseNovice** [Cyfrin/foundry-full-course-cu#3003] [Q&A] [answered]
Hey Team,

Every time I try to run the command foundryup-zksync I get the below error. 

I am using brew on a silicon chip M2 2022 Mac Air.

I have tried upgrading my terminal with the ar64 as some errors and help forums & chatgpt said but I seem to be going in circles trying to get this working. Can someone help me please?

dyld[48910]: Library not loaded: /opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: <55CEFD02-5B6C-3AE2-8405-B96F31CE975B> /Users/jesselees/.foundry/bin/forge
  Reason: tried: '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/usr/local/Cellar/libusb/1.0.27/lib/libusb-1.0.0.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/Cellar/libusb/1.0.27/lib/libusb-1.0.0.dylib' (no such file), '/usr/local/Cellar/libusb/1.0.27/lib/libusb-1.0.0.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64'))
foundryup-zksync: command failed: /Users/jesselees/.foundry/bin/forge --version
foundryup-zksync: installed - 
dyld[48913]: Library not loaded: /opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: <DB17DE42-455C-375D-AA69-00FA7EB5D3D9> /Users/jesselees/.foundry/bin/cast
  Reason: tried: '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/opt/homebrew/opt/libusb/lib/libusb-1.0.0.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/usr/local/Cellar/libusb/1.0.27/lib/libusb-1.0.0.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/Cellar/libusb/1.0.27/lib/libusb-1.0.0.dylib' (no such file), '/usr/local/Cellar/libusb/1.0.27/lib/libusb-1.0.0.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64'))
foundryup-zksync: command failed: /Users/jesselees/.foundry/bin/cast --version
foundryup-zksync: installed - 
foundryup-zksync: done!

**Comment by EngrPips** [comment]
I am unsure what could be wrong here.

**Comment by JesseNovice** [comment, marked as answer]
For anyone who had this issue, the problem was brew was installed to a different path which wasn't using the ARM64 architecture so I needed to:

1.  uninstall it and force it to install at /opt/homebrew 

2. add this to the path export PATH="/opt/homebrew/bin:$PATH"

3. finally install this brew install libusb

**Reply by EngrPips** [reply]
Awesome! Thanks for the heads-up.
